### PR TITLE
chore: replace issue templates with forms

### LINKED ISSUE #6981: Issue reporting - usage of issue forms
### Area for Improvement (required)

The area that could be improved are the GitHub issues namely switching from the markdown templates to the form-based templates.

### Observed behavior (required)

The repository contains markdown temaplates for the GitHub issues. This already supports the reporting as a basic structure is available. However, the layout could be improved and the validations (mandatory vs. optional areas) are not possible

### Desired behavior (required)

Usage of form-based templates fo the issue ([link](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)) that can improve the above mentioned areas and give more options for future improvements (like checkboxes or drop downdown menus

### Proposed Fix (optional)

Transfer issue templates form markdown into form/yaml-based templates

### rad Version (required)

n/a

### Operating system (required)

n/a

## Additional context

Also beeing a beta feature, I am using it since the public beta phase and did not face any issues since then. 

### Layout examples

Here are some screenshot how the new layout would look like. The first selection screen would remain unchanged:

overview

After selecting one the types the form-based templates is displayed. 

#### Bug template
Here is an example of the layout for bugs with a more concise strcuture for the reporter as well as validation i.e., obligatory fields are marked:

bug-form

There is also the option to have additional features like checkboxes:

bug-form2

#### Engineering improvement

Example for the engineering improvement (upper area):

engineering_1

#### Feature request

Example for feature request:

feature

### Testing

You can test it in the repository <https://github.com/lechnerc77/ghaction-testing/issues/new/choose> to see the validation options etc.

[AB#10805](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10805)

### PR DESCRIPTION
**Pull request by lechnerc77**
# Description

This PR contains a proposal for the migration of the issue templates from markdown to form-based to improve the usability for reporters. For details and screenshots of the layout, please check issue #6981 

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #6981

## Auto-generated summary

copilot:all

### CODE REVIEW

**Comment by rynowak** [comment]
Hi @lechnerc77 - thanks for trying this out. I hope you don't mind if we put this on pause for a week or so to give folks a chance to discuss. Most of the team is celebrating the holidays right now. 

This mostly affects the people who review and prioritize incoming issues so I want to make sure they can give their input.

**Comment by lechnerc77** [comment]
@rynowak all fine, I would not have expected any feedback in the holiday season.

File: .github/ISSUE_TEMPLATE/bug.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  - type: textarea
+    attributes:
+      label: Workaround
+      description: Have you found a workaround to get you unblocked?
+    validations:
+      required: false
+  - type: markdown
+    attributes:
+      value: "## System information"
+  - type: input
```
Comment (AaronCrawfis) [inline .github/ISSUE_TEMPLATE/bug.yaml:40]: `rad version` will have a multi-line output. A `textarea` input would probably be better here.
```suggestion
  - type: textarea
```

File: .github/ISSUE_TEMPLATE/engineering.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  - type: textarea
+    attributes:
+      label: Proposed Fix
+      description: Have you found a way to implement or fix the issue?
+    validations:
+      required: false
+  - type: markdown
+    attributes:
+      value: "## System information"
+  - type: input
```
Comment (AaronCrawfis) [inline .github/ISSUE_TEMPLATE/engineering.yaml:40]:
```suggestion
  - type: textarea
```

File: .github/ISSUE_TEMPLATE/bug.yaml (hunk 1 above)
Comment (lechnerc77) [reply .github/ISSUE_TEMPLATE/bug.yaml:40]: Oh yes, that's right.

**Comment by lechnerc77** [comment]
@AaronCrawfis thanks for your suggestions. I added them to the PR.

**Review by AaronCrawfis — APPROVED** [review]
Looks great, thank! I checked out the experience over at https://github.com/lechnerc77/radius/tree/chore/issue-forms/.github/ISSUE_TEMPLATE and they all look great

**Comment by lechnerc77** [comment]
@AaronCrawfis anything else you need from my side efore you can merge the PR?

**Comment by rynowak** [comment]
> @AaronCrawfis anything else you need from my side efore you can merge the PR?

I'll help get this merged.

**Review by rynowak — APPROVED** [review]

**Comment by rynowak** [comment]
Thanks @lechnerc77 !

**Pull request merged**